File: haft-dashboard/scripts/deploy.py
#!/usr/bin/env python3
"""
FTP Deployment Script
Env vars from .env.deploy:
- FTP_HOST=82.29.157.61
- FTP_USER=u212019412.adxengine.net
- FTP_PASSWORD=***
- FTP_REMOTE_PATH=/home/u212019412/domains/adxengine.net/public_html/haft
"""

import os
import sys
from pathlib import Path
from ftplib import FTP, all_errors
from dotenv import load_dotenv

# Load .env.deploy
load_dotenv('.env.deploy')

FTP_HOST = os.getenv('FTP_HOST')
FTP_PORT = int(os.getenv('FTP_PORT', 21))
FTP_USER = os.getenv('FTP_USER')
FTP_PASSWORD = os.getenv('FTP_PASSWORD')
FTP_REMOTE_PATH = os.getenv('FTP_REMOTE_PATH')

def upload_dir(ftp, local_dir, remote_dir):
    """Recursively upload directory"""
    for item in Path(local_dir).rglob('*'):
        if item.is_file():
            rel_path = item.relative_to(local_dir)
            remote_file = f"{remote_dir}/{rel_path}".replace('\\', '/')
            
            # Create remote dirs
            remote_parent = '/'.join(remote_file.split('/')[:-1])
            try:
                ftp.cwd(remote_parent)
                ftp.cwd(FTP_REMOTE_PATH)
            except:
                pass
            
            print(f"  📤 {remote_file}")
            with open(item, 'rb') as f:
                ftp.storbinary(f'STOR {remote_file.split("/")[-1]}', f)

def deploy():
    print(f"🚀 Deploying to {FTP_HOST}:{FTP_PORT}")
    print(f"📁 Remote: {FTP_REMOTE_PATH}")
    
    try:
        ftp = FTP()
        ftp.connect(FTP_HOST, FTP_PORT, timeout=30)
        ftp.login(FTP_USER, FTP_PASSWORD.strip("'"))
        print("✅ Connected")
        
        # Ensure remote dir exists
        try:
            ftp.cwd(FTP_REMOTE_PATH)
        except:
            # Create path if needed
            ftp.cwd('/')
            for part in FTP_REMOTE_PATH.strip('/').split('/'):
                try:
                    ftp.cwd(part)
                except:
                    ftp.mkd(part)
                    ftp.cwd(part)
        
        # Upload .next
        print("📤 Uploading .next...")
        next_dir = Path('.next')
        if next_dir.exists():
            for file in next_dir.rglob('*'):
                if file.is_file():
                    rel_path = file.relative_to('.')
                    try:
                        ftp.storbinary(f'STOR {rel_path}', open(file, 'rb'))
                    except Exception as e:
                        print(f"  ⚠️  {rel_path}: {e}")
        
        # Upload public
        print("📤 Uploading public...")
        public_dir = Path('public')
        if public_dir.exists():
            for file in public_dir.rglob('*'):
                if file.is_file():
                    rel_path = file.relative_to('.')
                    try:
                        ftp.storbinary(f'STOR {rel_path}', open(file, 'rb'))
                    except:
                        pass
        
        # Upload key files
        for f in ['package.json', 'next.config.ts', '.env.local']:
            if Path(f).exists():
                print(f"📤 {f}")
                try:
                    with open(f, 'rb') as file:
                        ftp.storbinary(f'STOR {f}', file)
                except:
                    pass
        
        ftp.quit()
        print("✨ Deploy complete! Visit https://haft.adxengine.net")
        
    except all_errors as e:
        print(f"❌ FTP Error: {e}")
        sys.exit(1)

if __name__ == '__main__':
    deploy()

File: haft-dashboard/scripts/deploy.js
#!/usr/bin/env node
/**
 * FTP Deployment Script for HAFT Dashboard
 * 
 * Environment Variables (from .env.deploy):
 * - FTP_HOST: 82.29.157.61
 * - FTP_PORT: 21
 * - FTP_USER: u212019412.adxengine.net
 * - FTP_PASSWORD: [stored in .env.deploy]
 * - FTP_REMOTE_PATH: /home/u212019412/domains/adxengine.net/public_html/haft
 * 
 * Usage: node scripts/deploy.js
 */

const FTP = require('basic-ftp');
const fs = require('fs');
const path = require('path');
require('dotenv').config({ path: '.env.deploy' });

async function deploy() {
  const client = new FTP.Client();
  
  try {
    // Validate environment variables
    const required = ['FTP_HOST', 'FTP_PORT', 'FTP_USER', 'FTP_PASSWORD', 'FTP_REMOTE_PATH'];
    for (const env of required) {
      if (!process.env[env]) {
        throw new Error(`Missing required env var: ${env} (check .env.deploy)`);
      }
    }

    console.log(`🚀 Deploying to ${process.env.FTP_HOST}:${process.env.FTP_PORT}`);
    console.log(`📁 Remote path: ${process.env.FTP_REMOTE_PATH}`);

    await client.access({
      host: process.env.FTP_HOST,
      port: parseInt(process.env.FTP_PORT),
      user: process.env.FTP_USER,
      password: process.env.FTP_PASSWORD,
      secureOptions: { rejectUnauthorized: false },
    });

    console.log('✅ Connected to FTP server');

    // Upload .next directory
    const nextDir = path.join(__dirname, '../.next');
    const publicDir = path.join(__dirname, '../public');

    if (!fs.existsSync(nextDir)) {
      throw new Error('.next directory not found. Run `npm run build` first.');
    }

    // Change to remote directory
    try {
      await client.cd(process.env.FTP_REMOTE_PATH);
    } catch (e) {
      console.log(`⚠️  Remote directory doesn't exist, creating...`);
      await client.ensureDir(process.env.FTP_REMOTE_PATH);
    }

    // Upload .next
    console.log('📤 Uploading .next...');
    await client.uploadDir(nextDir, '.next');
    console.log('✅ .next uploaded');

    // Upload public if it exists
    if (fs.existsSync(publicDir)) {
      console.log('📤 Uploading public...');
      await client.uploadDir(publicDir, 'public');
      console.log('✅ public uploaded');
    }

    // Upload key files
    const filesToUpload = ['package.json', 'next.config.ts', '.env.local'];
    for (const file of filesToUpload) {
      const filePath = path.join(__dirname, '../', file);
      if (fs.existsSync(filePath)) {
        console.log(`📤 Uploading ${file}...`);
        await client.uploadFile(filePath, file);
      }
    }

    console.log('✅ All files uploaded');
    await client.close();
    console.log('✨ Deployment complete!');
  } catch (err) {
    console.error('❌ Deployment failed:', err.message);
    process.exit(1);
  }
}

deploy();
